echopype/calibrate/env_params.py

```diff
@@ -59,199 +68,289 @@ def harmonize_env_param_time(
+        A list of channels to be calibrated.
+        For EK80 data, this list has to corresponds with the subset of channels
+        selected based on waveform_mode and encode_mode
 
     Returns
     -------
-    A dictionary containing the calibration parameters.
-    """
-    out_dict = {}
+    dict
+        A dictionary containing sanitized user-provided environmental parameters.
 
-    # Temperature comes from either user input or data file
-    out_dict["temperature"] = user_env_dict.get(
-        "temperature", echodata["Environment"]["temperature"]
-    )
+    Notes
+    -----
+        The user-provided 'sound_absorption' parameter has to be a list or an xr.DataArray,
+        because this parameter is frequency-dependen.
+    """
+    # Make channel a sorted list
+    if not isinstance(channel, (list, xr.DataArray)):
+        raise ValueError("'channel' has to be a list or an xr.DataArray")
 
-    # Salinity and pressure always come from user input
-    if ("salinity" not in user_env_dict) or ("pressure" not in user_env_dict):
-        raise ReferenceError("Please supply both salinity and pressure in env_params.")
+    if isinstance(channel, xr.DataArray):
+        channel_sorted = sorted(channel.values)
     else:
-        out_dict["salinity"] = user_env_dict["salinity"]
-        out_dict["pressure"] = user_env_dict["pressure"]
-
-    # Always calculate sound speed and absorption
-    out_dict["sound_speed"] = uwa.calc_sound_speed(
-        temperature=out_dict["temperature"],
-        salinity=out_dict["salinity"],
-        pressure=out_dict["pressure"],
-        formula_source="AZFP",
-    )
-    out_dict["sound_absorption"] = uwa.calc_absorption(
-        frequency=echodata["Sonar/Beam_group1"]["frequency_nominal"],
-        temperature=out_dict["temperature"],
-        salinity=out_dict["salinity"],
-        pressure=out_dict["pressure"],
-        formula_source="AZFP",
-    )
-
-    # Harmonize time coordinate between Beam_groupX (ping_time) and env_params (time1)
-    # Note for AZFP data is always in Sonar/Beam_group1
-    for p in out_dict.keys():
-        out_dict[p] = harmonize_env_param_time(
-            out_dict[p], ping_time=echodata["Sonar/Beam_group1"]["ping_time"]
-        )
+        channel_sorted = sorted(channel)
+
+    # Screen parameters: only retain those defined in ENV_PARAMS
+    #  -- transform params in list to xr.DataArray
+    #  -- directly pass through params that are scalar or str
+    #  -- check channel coordinate if params are xr.DataArray and pass it through
+    out_dict = dict.fromkeys(ENV_PARAMS)
+    for p_name, p_val in user_dict.items():
+        if p_name in out_dict:
+            # Param "sound_absorption" has to be an xr.DataArray or a list because it is freq-dep
+            if p_name == "sound_absorption" and not (p_val, (xr.DataArray, list)):
+                raise ValueError(
+                    "The 'sound_absorption' parameter has to be an xr.DataArray, "
```

Comment:
So, does it have to be an xr.DataArray or either it or a list, per the comment above?

Reply:
Yes, I forgot to change this.
```suggestion
                    "The 'sound_absorption' parameter has to be a list or an xr.DataArray, "
```